internal/agent/device/device.go

```diff
@@ -199,8 +199,13 @@ func (a *Agent) syncDevice(ctx context.Context) (bool, error) {
 	}
 
 	if desired.Os != nil {
+		bootedImage, err := a.osImageController.GetCurrentBootcImage(ctx)
```

Comment:
as I noted the controller should not expose logic for reuse. If the device needs access to a bootc client then give it one. The reason is we don't want to create cross dependencies on controllers. A controllers only job is to reconcile the spec that it owns.